packages/flutter/test/material/dropdown_test.dart

```diff
@@ -1102,6 +1107,311 @@ void main() {
     expect(enabledHintBox.size, equals(disabledHintBox.size));
   });
 
+  testWidgets('hint displays when the items list is empty, items is null, and disabledHint is null', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+
+    Widget build({ List<String> items }){
+      return buildFrame(
+        items: items,
+        buttonKey: buttonKey,
+        value: null,
+        hint: const Text('hint used when disabled'),
+        disabledHint: null,
+      );
+    }
+    // [hint] should display when [items] is null and [disabledHint] is not defined
```

Comment:
According to https://api.flutter.dev/flutter/material/DropdownButton/hint.html, the hint is displayed if value is null. 

There seems to be some pre-existing confusion about this. See https://api.flutter.dev/flutter/material/DropdownButton/value.html. Can you make the docs consistent about when the hint is displayed, and what exactly makes the button enabled/disabled?

Reply:
I've updated the docs to be consistent with what exactly makes a button enabled/disabled.